astropy/time/formats.py

```diff
@@ -569,10 +569,74 @@ def value(self):
                              flags=['refs_ok'],
                              op_dtypes=7*[iys.dtype] + [np.object])
         for iy, im, id, ihr, imin, isec, ifracsec, out in iterator:
-            out[...] = datetime(iy, im, id, ihr, imin, isec, ifracsec)
+            out[...] = datetime.datetime(iy, im, id, ihr, imin, isec, ifracsec)
```

Comment:
That explains my confusion. I think there were a few pieces of competing advice in the issue and PR that led to that convoluted code.